tests/Compiler.php

```diff
@@ -207,6 +207,440 @@ public static function unicodeMaps($out_dir)
+     */
+    private static \$mapped;
+
+    /**
+     * @var array<int, bool>
+     */
+    private static \$ignored;
+
+    /**
+     * @var array<int, string>
+     */
+    private static \$deviation;
+
+    /**
+     * @var array<int, bool>
+     */
+    private static \$disallowed;
+
+    /**
+     * @var array<int, string>
+     */
+    private static \$disallowed_STD3_mapped;
+
+    /**
+     * @var array<int, bool>
+     */
+    private static \$disallowed_STD3_valid;
+
+    /**
+     * @var bool
+     */
+    private static \$dataLoaded = false;
+
+    /**
+     * @param int \$codePoint
+     * @param bool \$useSTD3ASCIIRules
+     *
+     * @return array{status: string, mapping?: string}
+     */
+    public static function lookup(\$codePoint, \$useSTD3ASCIIRules)
+    {
+        if (!self::\$dataLoaded) {
+            self::\$dataLoaded = true;
+            self::\$mapped = require __DIR__ . '/mapped.php';
+            self::\$ignored = require __DIR__ . '/ignored.php';
+            self::\$deviation = require __DIR__ . '/deviation.php';
+            self::\$disallowed = require __DIR__ . '/disallowed.php';
+            self::\$disallowed_STD3_mapped = require __DIR__ . '/disallowed_STD3_mapped.php';
+            self::\$disallowed_STD3_valid = require __DIR__ . '/disallowed_STD3_valid.php';
+        }
+
+        if (isset(self::\$mapped[\$codePoint])) {
+            return array('status' => 'mapped', 'mapping' => self::\$mapped[\$codePoint]);
+        }
+
+        if (isset(self::\$ignored[\$codePoint])) {
+            return array('status' => 'ignored');
+        }
+
+        if (isset(self::\$deviation[\$codePoint])) {
+            return array('status' => 'deviation', 'mapping' => self::\$deviation[\$codePoint]);
+        }
+
+        if (isset(self::\$disallowed[\$codePoint])) {
+            return array('status' => 'disallowed');
+        }
+
+        \$isDisallowedMapped = isset(self::\$disallowed_STD3_mapped[\$codePoint]);
+
+        if (\$isDisallowedMapped || isset(self::\$disallowed_STD3_valid[\$codePoint])) {
+            \$status = 'disallowed';
+
+            if (!\$useSTD3ASCIIRules) {
+                \$status = \$isDisallowedMapped ? 'mapped' : 'valid';
+            }
+
+            if (\$isDisallowedMapped) {
+                return array('status' => \$status, 'mapping' => self::\$disallowed_STD3_mapped[\$codePoint]);
+            }
+
+            return array('status' => \$status);
+        }
+
+        // fall back to range checking for "disallowed"
+{$rangeFallback}
+
+        return array('status' => 'valid');
+    }
+
+    /**
+     * @codeCoverageIgnore
+     */
+    private function __construct()
+    {
+    }
+}
+
+CP_STATUS;
+
+        file_put_contents($out_dir . 'CodePointStatus.php', $s);
+    }
+
+    public static function idnViramaMap($out_dir)
+    {
+        $handle = fopen(self::getFile('DerivedCombiningClass.txt'), 'r');
+        $virama = [];
+
+        while (($line = fgets($handle)) !== false) {
```

Comment:
`while (false !== $line = fgets($handle)) {`

there are other CS issues, could you please run `php-cs-fix fix src/Intl/Idn` (and on the other patched files also)

Reply:
Just to double check, running `php-cs-fix fix src/Intl/Idn` wants to remove all the \ in front of global functions. Is that really desirable?

Reply:
yep, I confirm: they provide no perf benefits, except in a few special cases, which php-cs-fixer knows about.